e2e_test/e2e_test.go

```diff
@@ -67,6 +71,86 @@ var _ = g.Describe("Dpu side", g.Ordered, func() {
 	})
 
 	g.AfterEach(func() {
+
+		if restoreDpuOperatorConfig != nil {
+			config := configv1.DpuOperatorConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: crNamespace,
+					Name:      crName,
+				},
+				Spec: restoreDpuOperatorConfig.Spec,
+			}
+			restoreDpuOperatorConfig = nil
+			err := dpuSideClient.Create(context.TODO(), &config)
+			Expect(err).NotTo(HaveOccurred())
+			config = configv1.DpuOperatorConfig{}
+			Eventually(func() error {
+				return dpuSideClient.Get(context.TODO(), types.NamespacedName{Namespace: crNamespace, Name: crName}, &config)
+			}, testutils.TestAPITimeout, testutils.TestRetryInterval).Should(Succeed())
+		}
+
+	})
+
+	g.It("Should run Webhook to validate DpuOperatorConfig", func() {
+		var err error
+		var config configv1.DpuOperatorConfig
+
+		dpuOperatorConfig0 := &configv1.DpuOperatorConfig{}
+		err = dpuSideClient.Get(context.TODO(), types.NamespacedName{Namespace: crNamespace, Name: crName}, dpuOperatorConfig0)
+		if err == nil {
+			restoreDpuOperatorConfig = dpuOperatorConfig0
+			dpuSideClient.Delete(context.TODO(), dpuOperatorConfig0)
+		}
+
+		config = configv1.DpuOperatorConfig{}
+		Eventually(func() error {
+			return dpuSideClient.Get(context.TODO(), types.NamespacedName{Namespace: crNamespace, Name: crName}, &config)
+		}, testutils.TestAPITimeout, testutils.TestRetryInterval).ShouldNot(Succeed())
+
+		config = configv1.DpuOperatorConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: crNamespace,
+				Name:      "invalidname",
+			},
+			Spec: configv1.DpuOperatorConfigSpec{
+				Mode:     "host",
+				LogLevel: 2,
+			},
+		}
+		err = dpuSideClient.Create(context.TODO(), &config)
+		Expect(err).To(HaveOccurred())
+
+		config = configv1.DpuOperatorConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: crNamespace,
+				Name:      crName,
+			},
+			Spec: configv1.DpuOperatorConfigSpec{
+				Mode:     "invalidmode",
+				LogLevel: 2,
+			},
+		}
+		err = dpuSideClient.Create(context.TODO(), &config)
+		Expect(err).To(HaveOccurred())
+
+		config = configv1.DpuOperatorConfig{
```

Comment:
Same here, don't reuse the same var name. Use names like `goodConfig`, `badConfig`, `invalidConfig`